### DISCUSSION TITLE
TNFS support (Spectranet, FujiNet)

### DISCUSSION THREAD

**Discussion by 48kRAM** [retronas/discussions#111] [Ideas]
I think it would be useful if this project integrated the Tiny NFS daemon (tnfsd) that is used by the Spectranet and FujiNet projects.

http://spectrum.alioth.net/doc/index.php/TNFS_server

https://fujinet.online/

The server exists for Linux, including Raspberry Pi, so hopefully this won't be a lot of work.

Thanks!

-Josh (@48kRAM)

**Comment by danmons** [comment]
Thank you very much for this.  I'll add it to the ideas page and get it working as soon as I can.

**Comment by danmons** [comment]
Hi @48kRAM ,

Code has been committed.  The installer builds tnfsd from source, configures it to run as a service, and sets the service to auto-start on system boot.

Documentation shell is here, but is missing the client-side documentation, as I don't have the hardware to test:
https://github.com/danmons/retronas/wiki/TNFS

Are you in a position where you can test this, and potentially contribute some client-side documentation?  Happy for you to update this thread, and I can copy/paste into the wiki.

**Reply by sairuk** [reply]
Awesome I started looking at this last night then your commit came through.

**Comment by 48kRAM** [comment]
Install ran flawlessly on my Pi 400 (fresh 64-bit Debian 11 install). FujiNet on my Atari 800XL runs great against the RN TNFS service. Loading games and demos without issue.

Thanks for supporting this so quickly! There's really nothing much to document that isn't already documented on fujinet.online. The most relevant link, I suspect, would be this one: https://github.com/FujiNetWIFI/fujinet-platformio/wiki/FujiNet-Quickstart-Guide#tnfs-host-list which documents how to add a TNFS server to a server "slot" on the FujiNet. If you want screenshots of the process, I can capture from my XL.

**Comment by danmons** [comment]
Perfect thanks, I'll link to that install guide.  I don't want to double-up on work done if it's already there (nor do I want to host something that will just go stale).  

After your comments on Twitter and a bit more reading I think I've understood the different manufacturers and devices a bit better now, so I'll clean up the wiki entry and link to the individual pages and sites.

Thanks again for telling me about this.  Great to add another couple of old computers to the compatibility list.
